### PR TITLE
feat: AMQP / Ironic event source

### PR DESCRIPTION
**Pull request by skrobul**
This EventSource provides events generated by the in-cluster Ironic instance and can be consumed to react to changes happening there.

### CODE REVIEW

**Review by nicholaskuechler — APPROVED** [review]

File: components/argo-events/openstack-event-source.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+kind: EventSource
+metadata:
+  name: openstack-amqp
+spec:
+  amqp:
+    openstack:
+      # amqp server url
+      url: amqp://rabbitmq-server-0.rabbitmq-nodes.openstack.svc.cluster.local:5672/ironic
```
Comment (cardoe) [inline components/argo-events/openstack-event-source.yaml:8]: Can we only support 1 node here?

Comment (skrobul) [reply components/argo-events/openstack-event-source.yaml:8]: Unfortunately that seems to be the case - it did not work with comma separated, URL. I briefly looked into the code and it looks like Argo-Events simply [passes down](https://github.com/argoproj/argo-events/blob/d887e66f40dff3604f07755082758ba03c37204c/eventsources/sources/amqp/start.go#L109) the URL to the [amqp091 library's Dial function](https://pkg.go.dev/github.com/rabbitmq/amqp091-go#Dial) which does not seem to support anything but a single broker.

Additionally, the [project's non-goals](https://github.com/rabbitmq/amqp091-go?tab=readme-ov-file#non-goals) explicitly state that "Auto reconnect and re-synchronization of client and server topologies. " is not going to be supported.

Comment (cardoe) [reply components/argo-events/openstack-event-source.yaml:8]: Well that stinks.

Comment (skrobul) [reply components/argo-events/openstack-event-source.yaml:8]: Yeah. I didn't test this, but maybe we can create three instances of that EventSource, each with a different URL? But I don't know if there is an easy way to deduplicate events.

**Pull request merged**